Generated bindings attempt to derive `Default` for fixed length arrays over 32 length

### Component

Forge

### Have you ensured that all of these are up to date?

- [x] Foundry
- [x] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (1bb2d67 2024-02-19T00:17:47.001360000Z)

### What command(s) is the bug in?

`bind`

### Operating System

macOS (Apple Silicon)

### Describe the bug

The rust standard library derives `Default` for generic fixed lengths arrays length `0..=32`. See docs [here](https://doc.rust-lang.org/std/primitive.array.html#impl-Default-for-%5BT;+32%5D) and source code [here](https://doc.rust-lang.org/src/core/array/mod.rs.html#452).

I generated bindings for the [eigenlayer-middleware](https://github.com/Layr-Labs/eigenlayer-middleware) contracts. After generating a crate I ran `cargo build` and encountered the following error:


```sh
error[E0277]: the trait bound `[[u8; 32]; 44]: std::default::Default` is not satisfied
     |
3987 |         Default,
     |         ------- in this derive macro expansion
...
3993 |     pub struct GetHistoricalSummaryProofReturn(pub [[u8; 32]; 44]);
     |                                                ^^^^^^^^^^^^^^^^^^ the trait `std::default::Default` is not implemented for `[[u8; 32]; 44]`
     |
     = help: the following other types implement trait `std::default::Default`:
               [T; 0]
               [T; 1]
               [T; 2]
               [T; 3]
               [T; 4]
               [T; 5]
               [T; 6]
               [T; 7]
             and 27 others
     = note: this error originates in the derive macro `Default` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `[[u8; 32]; 46]: std::default::Default` is not satisfied
     |
4131 |         Default,
     |         ------- in this derive macro expansion
...
4137 |     pub struct GetValidatorProofReturn(pub [[u8; 32]; 46]);
     |                                        ^^^^^^^^^^^^^^^^^^ the trait `std::default::Default` is not implemented for `[[u8; 32]; 46]`
     |
     = help: the following other types implement trait `std::default::Default`:
               [T; 0]
               [T; 1]
               [T; 2]
               [T; 3]
               [T; 4]
               [T; 5]
               [T; 6]
               [T; 7]
             and 27 others
     = note: this error originates in the derive macro `Default` (in Nightly builds, run with -Z macro-backtrace for more info)
``` 

The binding generation logic should not attempt to derive `Default` for types that contains fixed length arrives > 32, and instead generate `Default` impls for those fixed length arrays.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -222,14 +222,14 @@ tower-http = "0.4"
 #ethers-solc = { path = "../ethers-rs/ethers-solc" }
 
 [patch.crates-io]
-ethers = { git = "https://github.com/gakonst/ethers-rs", rev = "f0e5b194f09c533feb10d1a686ddb9e5946ec107" }
-ethers-core = { git = "https://github.com/gakonst/ethers-rs", rev = "f0e5b194f09c533feb10d1a686ddb9e5946ec107" }
-ethers-contract = { git = "https://github.com/gakonst/ethers-rs", rev = "f0e5b194f09c533feb10d1a686ddb9e5946ec107" }
-ethers-contract-abigen = { git = "https://github.com/gakonst/ethers-rs", rev = "f0e5b194f09c533feb10d1a686ddb9e5946ec107" }
-ethers-providers = { git = "https://github.com/gakonst/ethers-rs", rev = "f0e5b194f09c533feb10d1a686ddb9e5946ec107" }
-ethers-signers = { git = "https://github.com/gakonst/ethers-rs", rev = "f0e5b194f09c533feb10d1a686ddb9e5946ec107" }
-ethers-middleware = { git = "https://github.com/gakonst/ethers-rs", rev = "f0e5b194f09c533feb10d1a686ddb9e5946ec107" }
-ethers-solc = { git = "https://github.com/gakonst/ethers-rs", rev = "f0e5b194f09c533feb10d1a686ddb9e5946ec107" }
+ethers = { git = "https://github.com/gakonst/ethers-rs", rev = "73e5de211c32a1f5777eb5194205bdb31f6a3502" }
+ethers-core = { git = "https://github.com/gakonst/ethers-rs", rev = "73e5de211c32a1f5777eb5194205bdb31f6a3502" }
+ethers-contract = { git = "https://github.com/gakonst/ethers-rs", rev = "73e5de211c32a1f5777eb5194205bdb31f6a3502" }
+ethers-contract-abigen = { git = "https://github.com/gakonst/ethers-rs", rev = "73e5de211c32a1f5777eb5194205bdb31f6a3502" }
+ethers-providers = { git = "https://github.com/gakonst/ethers-rs", rev = "73e5de211c32a1f5777eb5194205bdb31f6a3502" }
+ethers-signers = { git = "https://github.com/gakonst/ethers-rs", rev = "73e5de211c32a1f5777eb5194205bdb31f6a3502" }
+ethers-middleware = { git = "https://github.com/gakonst/ethers-rs", rev = "73e5de211c32a1f5777eb5194205bdb31f6a3502" }
+ethers-solc = { git = "https://github.com/gakonst/ethers-rs", rev = "73e5de211c32a1f5777eb5194205bdb31f6a3502" }
 
 revm = { git = "https://github.com/bluealloy/revm", branch = "reth_freeze" }
 revm-primitives = { git = "https://github.com/bluealloy/revm", branch = "reth_freeze" }

--- a/crates/forge/bin/cmd/bind.rs
+++ b/crates/forge/bin/cmd/bind.rs
@@ -77,6 +77,10 @@ pub struct BindArgs {
     #[clap(long)]
     skip_build: bool,
 
+    /// Don't add any additional derives to generated bindings
+    #[clap(long)]
+    skip_extra_derives: bool,
+
     #[clap(flatten)]
     build_args: CoreBuildArgs,
 }
@@ -161,10 +165,13 @@ impl BindArgs {
             })
             .map(|path| {
                 trace!(?path, "parsing Abigen from file");
-                Abigen::from_file(&path)
-                    .wrap_err_with(|| format!("failed to parse Abigen from file: {:?}", path))?
-                    .add_derive("serde::Serialize")?
-                    .add_derive("serde::Deserialize")
+                let abi = Abigen::from_file(&path)
+                    .wrap_err_with(|| format!("failed to parse Abigen from file: {:?}", path));
+                if !self.skip_extra_derives {
+                    abi?.add_derive("serde::Serialize")?.add_derive("serde::Deserialize")
+                } else {
+                    abi
+                }
             })
             .collect::<Result<Vec<_>, _>>()?;
         let multi = MultiAbigen::from_abigens(abigens).with_filter(self.get_filter());
@@ -207,11 +214,14 @@ No contract artifacts found. Hint: Have you built your contracts yet? `forge bin
 
     /// Generate the bindings
     fn generate_bindings(&self, artifacts: impl AsRef<Path>) -> Result<()> {
-        let bindings = self.get_multi(&artifacts)?.build()?;
+        let mut bindings = self.get_multi(&artifacts)?.build()?;
         println!("Generating bindings for {} contracts", bindings.len());
         if !self.module {
             trace!(single_file = self.single_file, "generating crate");
-            bindings.dependencies([r#"serde = "1""#]).write_to_crate(
+            if !self.skip_extra_derives {
+                bindings = bindings.dependencies([r#"serde = "1""#])
+            }
+            bindings.write_to_crate(
                 &self.crate_name,
                 &self.crate_version,
                 self.bindings_root(&artifacts),

